### PR TITLE
NIFI-3809 - Added HTTP mode and HTTP proxy for S2S Reporting Tasks

### PR DESCRIPTION
**Pull request by pvillard31**
Thank you for submitting a contribution to Apache NiFi.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [X] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [X] Does your PR title start with NIFI-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [X] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [X] Is your initial contribution a single, squashed commit?

### For code changes:
- [X] Have you ensured that the full suite of tests is executed via mvn -Pcontrib-check clean install at the root nifi folder?
- [X] Have you written or updated unit tests to verify your changes?
- [X] If adding new Properties, have you added .displayName in addition to .name (programmatic access) for each of the new properties?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Comment by ijokarumawak** [comment]
Reviewing...

File: nifi-nar-bundles/nifi-site-to-site-reporting-bundle/nifi-site-to-site-reporting-task/src/main/java/org/apache/nifi/reporting/AbstractSiteToSiteReportingTask.java [hunk 1]
```diff
@@ -73,25 +80,65 @@
...
             .build();
     static final PropertyDescriptor BATCH_SIZE = new PropertyDescriptor.Builder()
             .name("Batch Size")
+            .displayName("Batch Size")
             .description("Specifies how many records to send in a single batch, at most.")
             .required(true)
             .defaultValue("1000")
             .addValidator(StandardValidators.POSITIVE_INTEGER_VALIDATOR)
             .build();
+    static final PropertyDescriptor TRANSPORT_PROTOCOL = new PropertyDescriptor.Builder()
```
Comment (ijokarumawak) [inline nifi-nar-bundles/nifi-site-to-site-reporting-bundle/nifi-site-to-site-reporting-task/src/main/java/org/apache/nifi/reporting/AbstractSiteToSiteReportingTask.java]: If I am not missing anything, these newly added PropertyDescriptors should be used by sub classes such as SiteToSiteBulletinReportingTask?

**Comment by pvillard31** [comment]
Hey @ijokarumawak, thanks for the review! Good catch! This reporting task is not calling ``super.getSupportedPropertyDescriptors()``... I pushed another commit to address it, let me know if you want me to squash or if you have additional comments.

**Comment by ijokarumawak** [comment]
Thanks for the update @pvillard31 . I confirmed Proxy can be specified with all three reporting tasks. LGTM +1. I'm going to squash commits and merge into master. Thank you!

**Pull request closed without merging**